### PR TITLE
fix path join on windows

### LINKED ISSUE #53847: istioctl install - Error: generate config: helm render: load chart: component does not exist
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

Just starting using Istio latest version (1.24.0) on Windows 11. I followed the [Getting Started](https://istio.io/latest/docs/setup/getting-started) page.
When I run `istioctl install -f samples/bookinfo/demo-profile-no-gateways.yaml -y` or any `istioctl install`, I get the usual blue ship and "Error: generate config: helm render: load chart: component does not exist".

The error seems to come from [here](https://github.com/istio/istio/blob/34e975bd814ae44d81eaa7320769f3cf1d36fc42/operator/pkg/helm/helm.go#L155).

**The same command works perfectly on 1.23.3.** 

### Version

```Text
$ istioctl version
Istio is not present in the cluster: no running Istio pods in namespace "istio-system"
client version: 1.24.0
$ kubectl version --short
error: unknown flag: --short
See 'kubectl version --help' for usage.
$ helm version --short
v3.16.1+g5a5449d
```

### Additional Information

No logs

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

fixes: https://github.com/istio/istio/issues/53847

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
But we donot support windows

**Comment by zirain** [comment]
> But we donot support windows

istioctl support.

**Comment by hzxuzhonghu** [comment]
Ah good to know.  

But why `filepath.Join` doesnot support windows?

File: operator/pkg/helm/path.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// limitations under the License.
+
+package helm
+
+import (
+	"path/filepath"
+)
+
+func pathJoin(elem ...string) string {
+	return filepath.Join(elem...)
```
Comment (keithmattix) [inline operator/pkg/helm/path.go:24]: Shouldn't Go's Windows implementation of this function add the correct slashes?

Comment (howardjohn) [reply operator/pkg/helm/path.go:24]: I guess with fs.FS it uses forward slash while the go windows implementation assumes a real fs with backwards

**Review by howardjohn — APPROVED** [review]

**Pull request merged**